File: rwh/ch04/InteractWith.py
import sys 

def interact_with(func, input_file, output_file):
	with open(input_file, 'r') as input, open(output_file, 'w') as output:
		input = input.read()
		output.write(func(input))

def id_func(inp): return inp # identity function

if __name__ == '__main__':
	_, input_file, output_file = sys.argv
	interact_with(id_func, input_file, output_file)

File: rwh/ch04/InteractWith.scala
import scala.io.Source
import java.io.PrintWriter

def interactWith(func: String => String, inputFile: String, outputFile: String) = {
	val output = new PrintWriter(outputFile)
	val input = Source.fromFile(inputFile)
	output.write(input.mkString)
	input.close()
	output.close()
}

args match {
	case Array(inputFile, outputFile) => interactWith(identity, inputFile, outputFile)
	case _                            => println("error")
}